### PR TITLE
Rubocop todo fixes

### PR DESCRIPTION
**Pull request by ericproulx**
Hello,

Here are some fixes based on rubocop todo:
- Style/HashTransformValues
- Style/RescueStandardError
- Lint/RedundantRequireStatement
- Performance/RegexpMatch
- Style/MutableConstant
- Lint/DisjunctiveAssignmentInConstructor

### CODE REVIEW

File: lib/grape/validations/validators/coerce.rb [hunk 1]
```diff
@@ -76,7 +76,7 @@ def coerce_value(val)
         converter.call(val)
 
       # Some custom types might fail, so it should be treated as an invalid value
-      rescue
+      rescue StandardError
```
Comment (dblock) [inline lib/grape/validations/validators/coerce.rb]: This is actually a behavior change. It's the right one though. Let's add it to UPGRADING?

Comment (ericproulx) [reply lib/grape/validations/validators/coerce.rb]: Separate PR ?

Comment (dblock) [reply lib/grape/validations/validators/coerce.rb]: Probably best. Remove it from this one?

**Comment by dblock** [comment]
👍

**Pull request merged**